pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -354,6 +354,74 @@ func (p *staticPolicy) GetTopologyHints(s state.State, pod *v1.Pod, container *v
 	}
 }
 
+func (p *staticPolicy) getPodRequestedCPU(pod *v1.Pod) int {
+	// The maximum of requested CPUs by init containers.
+	requestedByInitContainers := 0
+	for _, container := range pod.Spec.InitContainers {
+		if _, ok := container.Resources.Requests[v1.ResourceCPU]; !ok {
+			continue
+		}
+		requestedCPU := p.guaranteedCPUs(pod, &container)
+		if requestedCPU > requestedByInitContainers {
+			requestedByInitContainers = requestedCPU
+		}
+	}
+	// The sum of requested CPUs by app containers.
+	requestedByAppContainers := 0
+	for _, container := range pod.Spec.Containers {
+		if _, ok := container.Resources.Requests[v1.ResourceCPU]; !ok {
+			continue
+		}
+		requestedByAppContainers += p.guaranteedCPUs(pod, &container)
+	}
+
+	requestedByPod := requestedByAppContainers
+	if requestedByInitContainers > requestedByAppContainers {
+		requestedByPod = requestedByInitContainers
+	}
+	klog.Errorf("[cpumanager] requestedByInits : %d, requestedByUsers : %d, requestedByPod : %d", requestedByInitContainers, requestedByAppContainers, requestedByPod)
+	return requestedByPod
+}
+
+func (p *staticPolicy) GetPodLevelTopologyHints(s state.State, pod *v1.Pod) map[string][]topologymanager.TopologyHint {
+	// Get a count of how many guaranteed CPUs have been requested by Pod.
+	requestedByPod := p.getPodRequestedCPU(pod)
+
+	if requestedByPod == 0 {
+		return nil
+	}
+
+	assignedCPUs := cpuset.NewCPUSet()
+	for _, container := range append(pod.Spec.InitContainers, pod.Spec.Containers...) {
```

Comment:
theoretically you can calculate `requestedByPod ` in this loop. Are the call to `guaranteedCPUs ` expensive to optimize for one loop?

Reply:
By keeping functions separate we can check in the beginning if pod is not in Guaranteed QoS and just skip it (checking if it already exists or has any resources allocated). 

> Are the call to guaranteedCPUs expensive to optimize for one loop?
I don't think it is needed to optimize it for one loop.

Reply:
I agree. Having these separated has made it much easier for me to confirm the flow of the logic here.